src/features/session_replay/aggregate/index.js

```diff
@@ -64,6 +64,8 @@ export class Aggregate extends AggregateBase {
      * -- When visibility changes from "hidden" -> "visible", it must capture a full snapshot for the replay to work correctly across tabs
     */
     this.hasSnapshot = false
+    /** Payload metadata -- Should indicate that the payload being sent has a meta node. The meta node should always precede a snapshot node. */
```

Comment:
Can you elaborate on the last sentence in this comment? Is "node" in reference the the network request or part of the object we send to the network.

Reply:
"Node" is such a non-descriptive word 😆  What I mean by this is, every time RRWeb emits an "event" (node as I called it above), it flows through our `store()` method. In practice, RRWeb always emits a "meta" node, then a "snapshot" node, but there is a chance that those two could be arbitrarily broken apart by the harvest scheduler, which could have some weird consequences (mostly viewport sizing issues) if the downstream API grabs the snapshot node that has no meta node. 

 This effort is to shove the "meta" node back into the payload if it detects a snapshot without a meta

Reply:
theres a (small, but real) chance it could split based on the interval timing (if the interval triggers a harvest after it got a meta node, but before it got a snapshot node), or based on the size (though super small here since the meta nodes are tiny)

Reply:
Thanks!